app/models/registrar.rb

```diff
@@ -175,6 +176,36 @@ def replace_nameservers(hostname, new_attributes, domains: [])
     end
   end
 
+  def add_nameservers(new_attributes, domains: [])
+    transaction do
+      domain_scope = domains.dup
+      domain_list = []
+      failed_list = []
+
+      return if domains.empty?
+
+      domain_scope.each do |domain_name|
+        domain = self.domains.find_by('name = ? OR name_puny = ?', domain_name, domain_name)
```

Comment:
How about putting that block in the separate method to lower the complexity/lines?